src/plugins/chart_expressions/expression_xy/public/components/split_chart.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useCallback } from 'react';
+import { GroupBy, SmallMultiples } from '@elastic/charts';
+import { ExpressionValueVisDimension } from '@kbn/visualizations-plugin/common';
+import { getColumnByAccessor, getFormatByAccessor } from '@kbn/visualizations-plugin/common/utils';
+import { Datatable } from '@kbn/expressions-plugin/public';
+import { FormatFactory } from '../types';
+
+interface SplitChartProps {
+  splitColumnAccessor?: ExpressionValueVisDimension | string;
+  splitRowAccessor?: ExpressionValueVisDimension | string;
+  columns: Datatable['columns'];
+  formatFactory: FormatFactory;
+}
+
+const SPLIT_COLUMN = '__split_column__';
+const SPLIT_ROW = '__split_row__';
+
+export const SplitChart = ({
+  splitColumnAccessor,
+  splitRowAccessor,
+  columns,
+  formatFactory,
+}: SplitChartProps) => {
+  const format = useCallback(
+    (value: unknown, accessor: ExpressionValueVisDimension | string) => {
+      const formatParams = getFormatByAccessor(accessor, columns);
+      const formatter = formatParams ? formatFactory(formatParams) : formatFactory();
+      return formatter.convert(value);
+    },
+    [columns, formatFactory]
+  );
+
+  const getData = useCallback(
+    (datum: Record<string, any>, accessor: ExpressionValueVisDimension | string) => {
+      const splitColumn = getColumnByAccessor(accessor, columns);
+      return datum[splitColumn!.id];
+    },
+    [columns]
+  );
+
+  return splitColumnAccessor || splitRowAccessor ? (
+    <>
+      {splitColumnAccessor && (
+        <GroupBy
+          id={SPLIT_COLUMN}
+          by={(spec, datum) => getData(datum, splitColumnAccessor)}
+          sort={'alphaAsc'}
```

Comment:
Shouldn't this use the data index? Otherwise we will run into issues with the agg based visualizations: https://github.com/elastic/kibana/blob/0a4f0fc5046943564cba3ba9952ede180fa352ae/src/plugins/vis_types/xy/public/chart_split.tsx#L35

Reply:
Totally makes sense. Got it. I haven't seen that implementation.